### ISSUE TITLE
proxy authentication not supported

### ISSUE THREAD

**Issue by EckeD** [msfreaks/EvergreenAdmx#10]
currently the script does not support proxy authentication:
![EvergreenAdmx-ProxyError](https://user-images.githubusercontent.com/31133224/141770935-23b32827-5e99-4e6d-826a-ccdade77f731.JPG)
I guess it should be as easy as using switch "-UseDefaultCredentials" on the Invoke-WebRequest executions, and "$listRequest.Credentials = [System.Net.CredentialCache]::DefaultCredentials" on the [Net.WebRequest] ones. Maybe optional if a specific global switch is provided, like "-UseProxyAuth" or so.

**Comment by msfreaks** [comment]
I don't have a proxy server anywhere to test this unfortunately.
How does this work with a scheduled task for local system for instance? or running using a service account?

**Comment by EckeD** [comment]
hi,

running both as Local System as well as with a service account works fine - as long as the proxy is configured to allow that. This is not something your script / tool must solve, but the proxy responsible has to do that (e.g. allow access for Domain Computers, which is Local System).

In the end it really boils down to allow an optional switch to provide the proxy credentials - and in most cases it should be enough to define the default credentials as I mentioned above.
The only place where this still fails unfortunately is for the Firefox ADMX, as those seem to be on FTP, and the FTP component does not support proxy auth.

So I would suggest to simply add an optional switch to the script to set the aforementioned default credentials switch, which should fix most proxy auth issues in company networks (it usually is uncommon to have to provide a specific account for proxy auth, rather than the currently logged-in one).
Something like "-UseProxyAuth", and if that switch is set, use the above switches on each of the WebRequest calls.

cheers

**Comment by msfreaks** [comment]
Can you check what this returns for you (if anything) please:
```powershell
# tests invoke-webrequest behind proxy
Invoke-WebRequest -Uri "https://msfreaks.wordpress.com" -Proxy ([System.Net.WebProxy]::GetDefaultProxy()) -ProxyCredential ([System.Net.CredentialCache]::DefaultNetworkCredentials)
```

Ideally test this in the context of the scheduled task for EvergreenAdmx (so local system, or a service account?).

If this works I will use that with a switch of -UseDefaultProxy.
If it doesn't we'll see what else can be done :)

Thanks!

**Comment by EckeD** [comment]
hey there,
Happy New Year at first ;)
Unfortunately this only works if you have a static proxy configured. In our case however we do not use that, but instead have a so-called proxy PAC file configured which dynamically assigns the correct proxy etc.
It would work if we used a static proxy.
But, as I stated above, it is much easier than you think :) the "Invoke-WebRequest" command does provide a switch paramter "-UseDefaultCredentials" - so the following works fine:
`Invoke-WebRequest -Uri "https://msfreaks.wordpress.com" -UseDefaultCredentials`
